tests/test_jointcal.py

```diff
@@ -174,6 +174,76 @@ def test_iterateFit_exceedMaxSteps(self):
         self.assertEqual(self.fitter.minimize.call_count, maxSteps)
         log.error.assert_called_with("testing failed to converge after %s steps" % maxSteps)
 
+    def test_moderate_chi2_increase(self):
+        """DM-25159: warn, but don't fail, on moderate chi2 increases between
+        steps.
+        """
+        chi2_1 = lsst.jointcal.chi2.Chi2Statistic()
+        chi2_1.chi2 = 100.0
+        chi2_1.ndof = 100
+        chi2_2 = lsst.jointcal.chi2.Chi2Statistic()
+        chi2_2.chi2 = 300.0
+        chi2_2.ndof = 100
+
+        chi2s = [self.goodChi2, chi2_1, chi2_2, self.goodChi2, self.goodChi2]
+        self.fitter.computeChi2.side_effect = chi2s
+        self.fitter.minimize.side_effect = [MinimizeResult.Chi2Increased,
+                                            MinimizeResult.Chi2Increased,
+                                            MinimizeResult.Chi2Increased,
+                                            MinimizeResult.Converged,
+                                            MinimizeResult.Converged]
+        with unittest.mock.patch.object(lsst.log.Log, "warn", wraps=self.jointcal.log.warn) as warn:
```

Comment:
This works of course, but there's also [`self.assertLogs`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertLogs). I would hope that works with `lsst.log.Log`, but am not sure.

Reply:
I'd forgotten that you can use the `assertLogs` context manager to check the contents of the log. We have to put it in a `with lsst.log.UsePythonLogging()` CM and use `assertIn`, but I think this might be a better approach. Thanks.